### PR TITLE
[feat](nereids) adjust min/max for partition key

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
after partition prune, adjust the min/max of partition key 
Issue Number: close #xxx

### CODE REVIEW

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run cloud_p0

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run cloud_p0

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java [hunk 1]
```diff
@@ -508,6 +520,125 @@ private Statistics computeOlapScan(OlapScan olapScan) {
...
+                                minExpr = key.getKeys().get(partitionColumnIdx);
+                                minValue = convertLegacyLiteralToDouble(minExpr);
+                                maxExpr = key.getKeys().get(partitionColumnIdx);
+                                maxValue = convertLegacyLiteralToDouble(maxExpr);
+                            } else {
+                                double current = convertLegacyLiteralToDouble(key.getKeys().get(partitionColumnIdx));
+                                if (current > maxValue) {
+                                    maxValue = current;
+                                    maxExpr = key.getKeys().get(partitionColumnIdx);
+                                } else if (current < minValue) {
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:564]: Should we use `else`? What if there is only one value in the ListPartitionItem items? In this case, either minValue or maxValue would not be set.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java [hunk 2]
```diff
@@ -508,6 +520,125 @@ private Statistics computeOlapScan(OlapScan olapScan) {
...
+    private ColumnStatistic updateMinMaxForListPartitionKey(OlapTable olapTable,
+            List<String> selectedPartitionNames,
+            SlotReference slot, ColumnStatistic cache) {
+        int partitionColumnIdx = olapTable.getPartitionColumns().indexOf(slot.getColumn().get());
+        if (partitionColumnIdx != -1) {
+            try {
+                LiteralExpr minExpr = null;
+                LiteralExpr maxExpr = null;
+                double minValue = 0;
+                double maxValue = 0;
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:547]: Should we use Double.POSITIVE_INFINITY and Double.NEGATIVE_INFINITY as the initial values?

**Pull request merged**